compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -593,7 +593,10 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             correct: if lifetimes_correct && args_correct {
                 Ok(())
             } else {
-                Err(GenericArgCountMismatch { reported: Some(ErrorReported), invalid_args })
+                Err(GenericArgCountMismatch {
+                    reported: Some(ErrorReported::new_i_know_what_im_doing()),
```

Comment:
Both `lifetimes_correct` and `args_correct` should be `Result<(), ErrorReported>`, and then you can combine them with `.and` instead of `&&`, and in the `Err` case you get a `ErrorReported` guaranteed.